gradle-plugins/compose/build.gradle.kts

```diff
@@ -69,21 +81,75 @@ dependencies {
 
     embedded(libs.download.task)
     embedded(libs.kotlin.poet)
+    hotReloadDep("hot-reload-gradle-plugin")
+    hotReloadDep("hot-reload-gradle-core")
+    hotReloadDep("hot-reload-gradle-idea")
+    hotReloadDep("hot-reload-core")
+    hotReloadDep("hot-reload-orchestration")
+    hotReloadDep("hot-reload-annotations-jvm")
     embedded(project(":preview-rpc"))
     embedded(project(":jdk-version-probe"))
 }
 
+
 val packagesToRelocate = listOf("de.undercouch", "com.squareup.kotlinpoet")
 
+val relocationPackage = "org.jetbrains.compose.internal"
+
+val hotReloadPackage = "org.jetbrains.compose.reload"
+
+val hotReloadPackageRelocated = "$relocationPackage.$hotReloadPackage"
+
+val hotReloadPropertiesPath = "META-INF/gradle-plugins/org.jetbrains.compose.hot-reload.properties"
+
+private class HotReloadPropertiesTransformer(hotReloadPackageRelocated: String) : com.github.jengelman.gradle.plugins.shadow.transformers.Transformer {
+    private val targetPath = "META-INF/gradle-plugins/org.jetbrains.compose.embedded.hot-reload.properties"
+    private val content = """
+        implementation-class=$hotReloadPackageRelocated.gradle.ComposeHotReloadPlugin
+    """.trimIndent()
+
+    override fun canTransformResource(element: FileTreeElement?): Boolean = false
+    override fun transform(context: TransformerContext?) = Unit
+    override fun hasTransformedResource(): Boolean = true
+
+    override fun modifyOutputStream(os: ZipOutputStream?, preserveFileTimestamps: Boolean) {
+        val entry = ZipEntry(targetPath)
+        entry.time = TransformerContext.getEntryTimestamp(preserveFileTimestamps, entry.time)
+        os?.run {
+            putNextEntry(entry)
+            write(content.toByteArray())
+            closeEntry()
+        }
+    }
+
+    override fun getName(): String = "Hot reload properties transformer"
+}
+
+fun ShadowJar.relocateHotReload() {
+    val relocator = object : SimpleRelocator(hotReloadPackage, hotReloadPackageRelocated, ArrayList(), ArrayList()) {
+        override fun canRelocatePath(path: String?): Boolean {
+            return super.canRelocatePath(path) &&
+                    // do not relocate orchestration as its objects are used in serialization.
+                    path?.startsWith("org/jetbrains/compose/reload/orchestration/") == false
```

Comment:
As of right now, we are locked out with this attempt: 
We cannot shade the orchestration as this breaks java.io.Serializable, we cannot embedd it either because of classpath conflicts